design/api_design.md

```diff
@@ -0,0 +1,64 @@
+# API design
+
+We will follow standard as possible for our REST APIs:
+
+- CRUD
+- Search API
+
+## Global
+
+For all APIs, a JSON will be returned:
+- in case of success, pertinent data depending of the API's purpose
+- in case of error, we will have the following parameters:
+    - `detail`: message that details the error
+
+## CRUD
+
+At least four APIs by database's table to Create, Read, Update and Delete data.
+
+## Search API
+
+At least one search API by database's table with at least pagination, filter and sorting.
+
+The pagination will be manage as following:
+
+- query parameters for pagination:
+    1. `page`: indicates the current page number
```

Comment:
I feel using offset is better than pages, and will result in less code in backend & frontend

Reply:
Using page instead of offset for pagination query parameters is largely a matter of usability and convention, and each approach has its own advantages.

![image](https://github.com/user-attachments/assets/03458aae-b65f-4440-9720-4b1f96db8af4)